warehouse/integrations/redshift/redshift.go

```diff
@@ -1439,12 +1338,30 @@ func (rs *Redshift) SetConnectionTimeout(timeout time.Duration) {
 	rs.connectTimeout = timeout
 }
 
+func (rs *Redshift) shouldMerge(isUsersTable bool) bool {
+	if !rs.config.allowMerge {
+		return false
+	}
+	if !rs.Uploader.CanAppend() {
```

Comment:
@cisse21 @achettyiitr @lvrach this will break backwards compatibility for hyperverge because previously on line 519 (master) we were just doing:

```go
if !slices.Contains(rs.config.skipDedupDestinationIDs, rs.Warehouse.Destination.ID) {
```

But now, in case `skipComputingUserLatestTraits` is `true` and `skipDedupDestinationIDs` contains the destination ID, we also check for the `uploader.CanAppend()`.

On the other hand `allowMerge` is fine because we don't set it so it's always `true`.

P.S.: I'm using the same `shouldMerge` logic in `postgres` as well even though there is no one with such configuration there. Let me know what you think.